### ISSUE TITLE
define a Decimal type, but old data(int32) will return NaN.

### ISSUE THREAD

**Issue by zysam** [Automattic/mongoose#6331]
mongoose: 5.0.12
mongoDB: >= 3.4

when schema has defined the path  Decimal128 type, and the old data type is number. it return Nan.

model.js
```javascript
onst mongoose = require('mongoose')

mongoose.connect('mongodb://localhost/test')

const schema = new mongoose.Schema({
  amount: {
    type: mongoose.Types.Decimal128,
    comment: 'test',
    get: function (val) { return Number(val) }
  }
}, {
  toJSON: { getters: true, virtuals: true },
  toObject: { getters: true, virtuals: true }
})

module.exports = mongoose.model('d1', schema)

```
main.js
```javascript
const D1 = require('./d1')

const main = async () => {

  const data = {
    amount: '11000'
  }
  let doc = await D1.create(data)
  doc = doc && doc.toJSON()
  console.log('doc', doc)
  console.log('number', Number(doc.amount))

  await D1.findOneAndUpdate({_id: doc._id}, {$inc: {amount: '5496.9'}})
  await D1.findOneAndUpdate({_id: doc._id}, {$inc: {amount: '-126.1'}})
  let newDoc = await D1.findById(doc._id)
  console.log('newDoc', newDoc.toJSON())

  let docs = await D1.find().sort('-_id')
  docs = docs.map(d => d.toJSON())
  console.log('docs =>', docs)
}

main()
```

node main.js

in d1 collection, amount already exists, and it is a int32 type, not a decimal type.
It will return amount: NaN

I have to fix db data to solve the problem. 
like this:
```javascript
Model.updateMany({}, {$inc: {amount: '0'})
```

So, Is it a bug?

**Comment by lineus** [comment]
I am able to confirm the behavior with the following repro scripts:

### 6331Populate.js
```js
#!/usr/bin/env node
'use strict';

const mongoose = require('mongoose');
mongoose.connect('mongodb://localhost/test');
const conn = mongoose.connection;
const Schema = mongoose.Schema;

const schema = new Schema({
  amount: Number
});

const Test = mongoose.model('test', schema);

const test = new Test({
  amount: 10000
});

async function run() {
  await conn.dropDatabase();
  await test.save();
  let n = await Test.count();
  console.log(`added ${n} documents.`);
  return conn.close();
}

run();
```
### 6331.js
```js
#!/usr/bin/env node
'use strict';

const mongoose = require('mongoose');
mongoose.connect('mongodb://localhost/test');
const conn = mongoose.connection;
const Schema = mongoose.Schema;

const schema = new Schema({
  amount: {
    type: mongoose.Types.Decimal128,
    comment: 'test',
    get: function (val) { return Number(val); }
  }
}, {
  toJSON: { getters: true, virtuals: true },
  toObject: { getters: true, virtuals: true }
});

const Test = mongoose.model('test', schema);

async function main() {
  let doc = await Test.findOne();
  console.log(doc);
  return conn.close();
}

main();
```
### output:
```
issues: ./6331Populate.js
added 1 documents.
issues: ./6331.js
{ _id: 5acf892f4577ef7d036d099b,
  __v: 0,
  amount: NaN,
  id: '5acf892f4577ef7d036d099b' }
issues:
```

**Comment by vkarpov15** [comment]
Closed by #6336
